<launch>
	<node name="amcl" pkg="amcl" type="amcl" >
	  <remap from="scan" to="pioneer1/scan_filtered" />
          <remap from="map_frame_id" to="map"/>
	  <param name="odom_frame_id" value="odom" />
	  <param name="base_frame_id" value="pioneer1/base_link" />
	  <param name="odom_model_type" value="diff" />
	  <param name="tf_broadcast" value="false" />
	  <param name="save_pose_rate" value="-1.0" />
	  <param name="recovery_alpha_slow" value="0.000" />
	  <param name="recovery_alpha_fast" value="0.0" />
	  <param name="initial_pose_x" value="0.0" />
	  <param name="initial_pose_y" value="0.0" />
	  <param name="initial_pose_a" value="0.0" />
	  <param name="initial_cov_xx" value="0.01*0.01" />
	  <param name="initial_cov_yy" value="0.01*0.01" />
	  <param name="initial_cov_aa" value="0.01*0.01*3.14*3.14/4" />
	  <param name="use_map_topic" value="true" />
	  <param name="laser_min_range" value="-1.0" />
	  <param name="laser_max_range" value="-1.0" />
	  <param name="gui_publish_rate" value="100.0" />
	  <param name="update_min_d" value="0.01" />
	  <param name="update_min_a" value="3.14/180" />
	  <param name="resample_interval" value="5" />

	 
    	  <param name="odom_alpha5" value="0.1"/>
	  <param name="min_particles" value="500"/>
    	  <param name="max_particles" value="2000"/>
    	  <!-- Maximum error between the true distribution and the estimated distribution. -->
    	  <param name="kld_err" value="0.05"/>
    	  <param name="kld_z" value="0.99"/>
    	  <param name="odom_alpha1" value="1.0"/>
    	  <param name="odom_alpha2" value="1.0"/>
    	  <!-- translation std dev, m -->
    	  <param name="odom_alpha3" value="2.0"/>
    	  <param name="odom_alpha4" value="1.0"/>
    	  <param name="laser_z_hit" value="1.0"/>
    	  <param name="laser_z_short" value="0.05"/>
    	  <param name="laser_z_max" value="0.05"/>
   	  <param name="laser_z_rand" value="0.5"/>
    	  <param name="laser_sigma_hit" value="0.2"/>
    	  <param name="laser_lambda_short" value="0.1"/>
    	  <param name="laser_model_type" value="likelihood_field"/>
    	  <!-- Maximum distance to do obstacle inflation on map, for use in likelihood_field model. -->
    	  <param name="laser_likelihood_max_dist" value="2.0"/>
    	  <!-- Translational movement required before performing a filter update.  -->
    	  <param name="update_min_d" value="0.3"/>
    	  <!--Rotational movement required before performing a filter update. -->
    	  <param name="update_min_a" value="0.314"/>
    	  <!-- Number of filter updates required before resampling. -->
    	  <param name="resample_interval" value="1"/>
    	  <!-- Increase tolerance because the computer can get quite busy -->
    	  <param name="transform_tolerance" value="1.0"/>
    	  <!-- Exponential decay rate for the slow average weight filter, used in deciding when to recover by adding random poses. A good value might be 0.001. -->
    	  <param name="recovery_alpha_slow" value="0.0"/>
    	  <!--Exponential decay rate for the fast average weight filter, used in deciding when to recover by adding random poses. A good value might be 0.1. -->
    	  <param name="recovery_alpha_fast" value="0.0"/>
          <!-- When set to true, AMCL will subscribe to the map topic rather than making a service call to receive its map.-->
    	  <param name="receive_map_topic" value="true"/>
    <!--  When set to true, AMCL will only use the first map it subscribes to, rather than updating each time a new one is received. -->
    	  <param name="first_map_only" value="false"/>
    
	</node>
	
	<!-- Visualizer-->
	<include file = "$(find munans_localization)/launch/Visualizer.launch" /> 

	<!-- Factor graph algorithm-->
	<node name = "factor_graph" pkg = "munans_localization" type = "factor_graph" respawn = "true" output="screen" />

	<!-- Autopilot-->
	<node name = "autopilot" pkg = "munans_localization" type = "autopilot"  output="screen" />

	<!-- Map server-->
	<node name = "map_server" pkg = "munans_localization" type = "map_server" respawn = "true" output="screen" />

	<!-- Accelerometer-->
	<node name = "accelerometer" pkg = "munans_localization" type = "accelerometer" respawn = "true" output="screen" />

	<!-- Odometer-->
	<node name = "odometer" pkg = "munans_localization" type = "odometer" respawn = "true" output="screen" />

	<!-- Ground truth-->
	<node name = "ground_truth" pkg = "munans_localization" type = "ground_truth" respawn = "true" output="screen" />

	<!-- Camera-->
	<node name = "camera" pkg = "munans_localization" type = "camera" respawn = "true" output="screen" />

	<!-- Factor graph solver-->
	<!--node name = "solver" pkg = "munans_localization" type = "solver" respawn = "true" output="screen" /-->
	
	<!-- Factor graph algorithm-->
	<node name = "place_recognizer" pkg = "munans_localization" type = "place_recognizer" respawn = "true" output="screen" /> 

	<!--<node name = "map" pkg = "map_server" type = "map_server" args = "$(find munans_localization)/src/maps/basement_correct_2020_02_17-2.yaml" /> -->
	<node name = "map" pkg = "map_server" type = "map_server" args = "$(find munans_localization)/src/maps/basement_2020_edited2.yaml" />

</launch>
